control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1622,6 +1641,37 @@ func (r *reconciler) ensureIngressControllersRemoved(ctx context.Context) (bool,
 		return false, fmt.Errorf("failed to force delete pods under openshift-ingress namespace: %w", errors.NewAggregate(errs))
 	}
 
+	// Remove ingress service and route that were created by HCCO in case of basedomain passthrough feature is enabled
+	if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform &&
+		hcp.Spec.Platform.Kubevirt != nil &&
+		hcp.Spec.Platform.Kubevirt.BaseDomainPassthrough != nil &&
+		*hcp.Spec.Platform.Kubevirt.BaseDomainPassthrough {
+		{
+			var namespace string
+			if hcp.Spec.Platform.Kubevirt.InfraKubeConfigSecret != nil && hcp.Spec.Platform.Kubevirt.InfraNamespace != "" {
+				namespace = hcp.Spec.Platform.Kubevirt.InfraNamespace
+			} else {
+				namespace = hcp.Namespace
+			}
+			cpService := manifests.IngressDefaultIngressPassthroughService(namespace)
+			cpPassthroughRoute := manifests.IngressDefaultIngressPassthroughRoute(namespace)
+
```

Comment:
note, that with my pr, you'll need to add the GenerateID to the names here in order to locate the service and route. Take a look at my creation logic.